src/main/java/cloud/eppo/BaseEppoClient.java

```diff
@@ -436,7 +480,12 @@ public JsonNode getJSONAssignment(
               subjectAttributes,
               EppoValue.valueOf(defaultValue.toString()),
               VariationType.JSON);
-      return parseJsonString(value.stringValue());
+      MapperNode mapperNode = parseJsonString(value.stringValue());
+      if (mapperNode != null) {
+        return ((JacksonMapperNode) mapperNode).getJsonNode();
```

Comment:
Would we want to stay away from Jackson-specific stuff here?

I vaguely recall one of the main reasons we used Jackson is because it could more easily differentiate between arrays and objects than Java's built-in JSON parsing. But if we code around that, we may be able to remove Jackson entirely. Unless it's much slower on parsing, which may matter for Android app initialization times. In which case, I think upstream-provided parser as you're doing here may be the move.